map_gen/maps/crash_site/commands.lua

```diff
@@ -47,14 +53,46 @@ callback =
     end
 )
 
+local entities_to_check = {
+    'spitter-spawner','biter-spawner',
+    'small-worm-turret', 'medium-worm-turret','big-worm-turret', 'behemoth-worm-turret', 
+    'small-spitter', 'medium-spitter', 'big-spitter', 'behemoth-spitter', 
+    'small-biter', 'medium-biter', 'big-biter', 'behemoth-biter'
+}
+
+local function map_cleared()
+
+    local get_entity_count = game.forces["enemy"].get_entity_count
+    for i = 1, #entities_to_check do
+        local name = entities_to_check[i]
+        if get_entity_count(name) > 0 then
+            return false
+        end
+    end
+    return true
+end
+
 local function restart(args, player)
     player = player or server_player
+    local sanitised_scenario = args.scenario_name
 
     if global_data.restarting then
         player.print('Restart already in progress')
         return
     end
 
+    if Rank.less_than(player.name, Ranks.admin) then
```

Comment:
I think this should be something like.
```suggestion
    if player ~= server_player and Rank.less_than(player.name, Ranks.admin) then
```
For the case of running this from the server console.